run_viralrecon.sh

```diff
@@ -0,0 +1,73 @@
+ #!/bin/bash
+# if any errors, then exit
+set -e
+
+###########################
+# Author: Pooja Gupta
+
+USAGE="
+Purpose: Bash script to run viralrecon bioinformatic pipeline with wastewater sequencing data.
+
+Usage: run_viralrecon.sh <wastewater sequencing run_name> | tee -a viralrecon.log
+
+Last updated on May 16,2023
+"
+###########################
+
+echo "$USAGE"
+run_name=$1
+
+script_dir='/Volumes/NGS/Bioinformatics/pooja/ww_analysis_scripts/Wastewater-genomic-analysis'
+analysis_dir='/Volumes/IDGenomics_NAS/wastewater_sequencing'
+mkdir -p /Volumes/IDGenomics_NAS/wastewater_sequencing/$run_name/analysis/viralrecon/work
+
+#Creating required folders for analysis
+ww_fastq=$analysis_dir/$run_name/raw_data/fastq
+out_dir=$analysis_dir/$run_name/analysis/viralrecon
+work_dir=$analysis_dir/$run_name/analysis/viralrecon/work
+results=$analysis_dir/$run_name/results
+
+infile=${run_name}_samplesheet.csv
+
+echo "$(date) : Run Wastewater sample data with viralrecon for run $run_name"
+echo "$(date) : First create input samplesheet for viralrecon pipeline"
+
+if [ ! -f "$out_dir/${run_name}_samplesheet.csv" ]
+    then
+        echo "$(date) : $infile does not exist. Creating samplesheet required to run viralrecon"
+        python $script_dir/conf-files/fastq_dir_to_samplesheet.py $ww_fastq $out_dir/$infile
+        echo "$(date) : $infile Samplesheet generated" 
+    else echo "$(date) : $infile already exists, starting viralrecon"
+fi
+
+#check R1 extension in the python script, if different from default
+
+echo "$(date) : Running viralrecon"
+#export SINGULARITY_CACHEDIR=/home/pgupta/singularity
+#export NXF_SINGULARITY_CACHEDIR=/home/pgupta/singularity
+
+#Use UPHL_viralrecon.config to update Pangolin container, if needed
+
+nextflow run nf-core/viralrecon --input $out_dir/$infile \
+                                --primer_set_version 5.3.2 \
+                                --outdir $out_dir \
+                                --nextclade_dataset false \
+				                --nextclade_dataset_tag false \
+                                --schema_ignore_params 'genomes,primer_set_version' \
+                                --multiqc_config $script_dir/conf-files/new_multiqc_config.yaml \
+                                -profile singularity \
+                                -params-file $script_dir/conf-files/UPHL_viralrecon_params.yml \
+                                -c $script_dir/conf-files/UPHL_viralrecon.config -w $work_dir
```

Comment:
Can you add '-resume-' to this command?

I ask because as I'm testing I get the following errors:
```
ERROR ~ Error executing process > 'NFCORE_VIRALRECON:ILLUMINA:PICARD_COLLECTMULTIPLEMETRICS (3)'

Caused by:
  Failed to pull singularity image
  command: singularity pull  --name depot.galaxyproject.org-singularity-picard-3.0.0--hdfd78af_1.img.pulling.1687545369491 https://depot.galaxyproject.org/singularity/picard:3.0.0--hdfd78af_1 > /dev/null
  status : 143
  message:
    INFO:    Downloading network image



 -- Check '.nextflow.log' file for details
```

This error means that there was an issue downloading the singularity image. This is generally resolved by rerunning things, but it'd be nice to not have to start from the beginning each time.

Reply:
I can add that.

Reply:
I have added the '-resume' option but I am not sure if I did it in the correct way. For me, I haven't been able to successfully use this option within a bash script for some reason.

Reply:
Ok, I read a bit more about using the '-resume' option. Give me a moment to make some more changes and then you can try re-running the command.

Reply:
I think it'll work. I'll try it the next time it fails.

Reply:
I just added in a better solution to the issue. I added '-resume' as an optional argument only when the pipeline fails after running viralrecon. The pipeline will not try to resume from a previous run if none exists. Now, when you are trying to resume the analysis, please run 
```
run_wwtp_sequencing_analysis.sh <wastewater sequencing run_name> -resume
```
The -resume flag is optional. If you don't provide it, $resume_option will be empty, and run_viralrecon.sh will run without the -resume flag, just like before. If you provide -resume, it will be passed to run_viralrecon.sh and Nextflow will attempt to resume the pipeline.